tools/wasi/uvwasi.cpp

```diff
@@ -0,0 +1,71 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2021 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "uvwasi.hpp"
+#include <uvwasi.h>
+
+namespace fizzy::wasi
+{
+namespace
+{
+class UVWASIImpl final : public UVWASI
+{
+    /// UVWASI state.
+    uvwasi_t m_state{};
+
+public:
+    ~UVWASIImpl() final { uvwasi_destroy(&m_state); }
```

Comment:
Apparently we'll need a flag to check if `uvwasi_init` was already called sucessfully.